onnxruntime/contrib_ops/cpu/transformers/generation_device_helper.cc

```diff
@@ -210,6 +211,21 @@ void InitBeamState(transformers::IBeamSearchState<T>* beam_state,
   }
 }
 
+template <typename T>
+void InitGreedyState(transformers::IGreedySearchState<T>* greedy_state,
+                     gsl::span<int32_t>& sequence_lengths,
+                     int batch_size,
+                     void* /*stream*/) {
+  memset(greedy_state->next_token_scores.data(), 0, greedy_state->next_token_scores.size_bytes());
+  memset(greedy_state->next_tokens.data(), 0, greedy_state->next_tokens.size_bytes());
+  memset(greedy_state->next_positions.data(), 0, greedy_state->next_positions.size_bytes());
+
+  gsl::span<int32_t>& next_position = greedy_state->next_positions;
+  for (int i = 0; i < batch_size; i++) {
+    next_position[SafeInt<gsl::index>(i)] = sequence_lengths[i] - 1;
```

Comment:
In BeamSearch, currently next_position is initialized as sequence_lengths. Shall we make them consistent?